pkg/planner/core/point_get_plan.go

```diff
@@ -370,23 +424,46 @@ func (p *PointGetPlan) PrunePartitions(sctx sessionctx.Context) bool {
 	//    and it does not have the PartitionIdx set
 	if !p.SCtx().GetSessionVars().StmtCtx.UseCache() &&
 		p.PartitionIdx != nil {
-		return false
+		return false, nil
 	}
 	is := sessiontxn.GetTxnManager(sctx).GetTxnInfoSchema()
 	tbl, ok := is.TableByID(context.Background(), p.TblInfo.ID)
 	if tbl == nil || !ok {
 		// Can this happen?
 		intest.Assert(false)
```

Comment:
It seems like this Assert can be removed since now we'll return an error in this case.